### PR TITLE
Added function to convert from decimal to another base

### PR DESCRIPTION
**Pull request by Kevin-Escobedo**
### **Describe your change:**
I added a new function that converts from decimal to any other base.

* [X] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [X] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [X] This pull request is all my own work -- I have not plagiarized.
* [X] I know that pull requests will not be merged if they fail the automated tests.
* [X] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [X] All new Python files are placed inside an existing directory.
* [X] All filenames are in all lowercase characters with no spaces or dashes.
* [X] All functions and variable names follow Python naming conventions.
* [X] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [X] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [X] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [X] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: conversions/decimal_to_any.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        raise TypeError("'float' object cannot be interpreted as an integer")
+
+    HEXADECIMAL = {'10': 'A', '11': 'B', '12': 'C', '13': 'D', '14': 'E', '15': 'F'}
+    new_value = ""
+    mod = 0
+    div = 0
+    if base == 0 or base == 1:
+        return
+    while ((div != 1) or (div != 0)):
+        mod = num % base
```
Comment (cclauss) [inline conversions/decimal_to_any.py:53]:
```suggestion
        div, mod = divmod(num, base)
```
https://docs.python.org/3/library/functions.html#divmod

File: conversions/decimal_to_any.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        >>> # negatives will error
+        >>> decimal_to_any(-45, 8) # doctest: +ELLIPSIS
+        Traceback (most recent call last):
+        ...
+        ValueError: parameter must be positive int
+        >>> # floats will error
+        >>> decimal_to_any(34.4, 6) # doctest: +ELLIPSIS
+        Traceback (most recent call last):
+        ...
+        TypeError: 'float' object cannot be interpreted as an integer
```
Comment (cclauss) [inline conversions/decimal_to_any.py:25]: The exception text should match the exception raised by `int(34.4, base=6)`
https://docs.python.org/3.4/library/functions.html?highlight=open#int

File: conversions/decimal_to_any.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        >>> decimal_to_any(0, 2)
+        '0'
+        >>> decimal_to_any(5, 4)
+        '11'
+        >>> decimal_to_any(20, 3)
+        '202'
+        >>> decimal_to_any(58, 16)
+        '3A'
+        >>> # negatives will error
+        >>> decimal_to_any(-45, 8) # doctest: +ELLIPSIS
```
Comment (cclauss) [inline conversions/decimal_to_any.py:17]:
```suggestion
        >>> decimal_to_any(-45, 8)  # doctest: +ELLIPSIS
```
For improved readability

File: conversions/decimal_to_any.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        >>> decimal_to_any(10, '16') # doctest: +ELLIPSIS
+        Traceback (most recent call last):
+        ...
+        TypeError: 'str' object cannot be interpreted as an integer
+    """
+    if isinstance(num, float):
+        raise TypeError("'float' object cannot be interpreted as an integer")
+    if num < 0:
+        raise ValueError("parameter must be positive int")
+    if type(base) == str:
```
Comment (cclauss) [inline conversions/decimal_to_any.py:41]:
```suggestion
    if isinstance(base, str):
```

File: conversions/decimal_to_any.py [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+        raise TypeError("'str' object cannot be interpreted as an integer")
+    if isinstance(base, float):
+        raise TypeError("'float' object cannot be interpreted as an integer")
+
+    HEXADECIMAL = {'10': 'A', '11': 'B', '12': 'C', '13': 'D', '14': 'E', '15': 'F'}
+    new_value = ""
+    mod = 0
+    div = 0
+    if base in (0, 1):
+        return
```
Comment (cclauss) [inline conversions/decimal_to_any.py:51]: Bad input should raise exceptions, not return None. 
```
>>> int('77', base=0)
77
>>> int('77', base=1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: int() base must be >= 2 and <= 36, or 0
```

File: conversions/decimal_to_any.py [hunk 6]
```diff
@@ -0,0 +1,82 @@
+"""Convert a positive Decimal Number to Any Other Representation"""
+
+
+def decimal_to_any(num: int, base: int) -> str:
+
+    """
+        Convert a Integer Decimal Number to a Binary Number as str.
```
Comment (cclauss) [inline conversions/decimal_to_any.py:7]:
```suggestion
        Convert positive integers to another base as str.
```

File: conversions/decimal_to_any.py [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+        >>> decimal_to_any(-45, 8)  # doctest: +ELLIPSIS
+        Traceback (most recent call last):
+        ...
+        ValueError: parameter must be positive int
+        >>> # floats will error
+        >>> decimal_to_any(34.4, 6) # doctest: +ELLIPSIS
+        Traceback (most recent call last):
+        ...
+        TypeError: 'float' object cannot be interpreted as an integer
+        >>> #a float base will error
```
Comment (cclauss) [inline conversions/decimal_to_any.py:26]:
```suggestion
        >>> # a float base will error
```

**Comment by TheSuperNoob** [comment]
This will only work on bases up to 16, so maybe we should guard against entering a base higher than 16?

**Comment by cclauss** [comment]
https://docs.python.org/3/library/functions.html?highlight=open#int goes up to base 32.

**Comment by TheSuperNoob** [comment]
> https://docs.python.org/3/library/functions.html?highlight=open#int goes up to base 32.

Yes, but this algorithm does not currently have support for any more symbols than base 16. (F) However you could add the rest of the alphabet to the `HEXADECIMAL` dictionary to add support up to 36 like pythons standard does.

**Comment by Kevin-Escobedo** [comment]
Good idea! I'll start working on that

**Comment by Kevin-Escobedo** [comment]
Adding support for conversions up to base-36 was actually much simpler than I thought it would be. Anyway, looking forward to your review

**Comment by cclauss** [comment]
@Kevin-Escobedo Thanks for this submission.  Very cool.  I think things look good up until base 11.
If we add these lines to the end of `__main__`...
```
    for base in range(2, 37):
        print(f"Testing {base = }...")
        for num in range(1000):
            assert int(decimal_to_any(num, base), base) == num, (
                num, base, decimal_to_any(num, base),
                int(decimal_to_any(num, base), base)
            )
```
...we get:
```
Testing base = 2...
Testing base = 3...
Testing base = 4...
Testing base = 5...
Testing base = 6...
Testing base = 7...
Testing base = 8...
Testing base = 9...
Testing base = 10...
Testing base = 11...
Traceback (most recent call last):
  File "decimal_to_any.py", line 105, in <module>
    assert int(decimal_to_any(num, base), base) == num, (
AssertionError: (10, 11, '01', 1)
```

**Comment by cclauss** [comment]
Another interesting test is:
```
    for base in range(2, 37):
        assert decimal_to_any(base, base) == '10', base  # fails at 16
```

**Comment by Kevin-Escobedo** [comment]
@cclauss Thanks for testing and pointing out the flaws in my code, I feel it's really helped improve the overall algorithm!

**Comment by Kevin-Escobedo** [comment]
@cclauss Wait, there was an error with the TravisCI build because of an issue with parentheses not having the same indentation, I tried to fix it, but the pull request was already approved, does it matter?

**Comment by cclauss** [comment]
No troubles...  psf/black will fix it automatically when a maiintainer runs the autoblack GitHub Action.

**Pull request merged**